utils/azExtResourceType.d.ts

```diff
@@ -0,0 +1,79 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+/**
+ * Normalized type for Azure resources that uniquely identifies resource type for the purposes
+ * of the Azure extensions
+ */
+export declare enum AzExtResourceType {
+    AppServices = 'AppServices',
+    StaticWebApps = 'StaticWebApps',
+    FunctionApp = 'FunctionApp',
+    LogicApp = 'LogicApp',
+    VirtualMachines = 'VirtualMachines',
+    StorageAccounts = 'StorageAccounts',
+    NetworkSecurityGroups = 'NetworkSecurityGroups',
+    LoadBalancers = 'LoadBalancers',
+    Disks = 'Disks',
+    Images = 'Images',
+    AvailabilitySets = 'AvailabilitySets',
+    VirtualMachineScaleSets = 'VirtualMachineScaleSets',
+    VirtualNetworks = 'VirtualNetworks',
+    FrontDoorAndCdnProfiles = 'FrontDoorAndCdnProfiles',
+    PublicIpAddresses = 'PublicIpAddresses',
+    NetworkInterfaces = 'NetworkInterfaces',
+    NetworkWatchers = 'NetworkWatchers',
+    BatchAccounts = 'BatchAccounts',
+    ContainerRegistry = 'ContainerRegistry',
+    PostgresqlServersStandard = 'PostgresqlServersStandard',
+    PostgresqlServersFlexible = 'PostgresqlServersFlexible',
+    MysqlServers = 'MysqlServers',
+    SqlDatabases = 'SqlDatabases',
+    SqlServers = 'SqlServers',
+    AzureCosmosDb = 'AzureCosmosDb',
+    OperationalInsightsWorkspaces = 'OperationalInsightsWorkspaces',
+    OperationsManagementSolutions = 'OperationsManagementSolutions',
+    ApplicationInsights = 'ApplicationInsights',
+    AppServicePlans = 'AppServicePlans',
+    AppServiceKubernetesEnvironment = 'AppServiceKubernetesEnvironment',
+    ContainerAppsEnvironment = 'ContainerAppsEnvironment',
+    ContainerApps = 'ContainerApps',
+
+    // Below are not supported but have icons in the Resources extension

```

Comment:
Assuming the meaning of "supported" here is that it shows up by default in the resources view, then most of these aren't supported.

By that definition, the supported ones are:

```typescript
AppServices = 'AppServices',
AzureCosmosDb = 'AzureCosmosDb',
ContainerAppsEnvironment = 'ContainerAppsEnvironment',
ContainerApps = 'ContainerApps',
FunctionApp = 'FunctionApp',
PostgresqlServersStandard = 'PostgresqlServersStandard',
PostgresqlServersFlexible = 'PostgresqlServersFlexible',
StaticWebApps = 'StaticWebApps',
StorageAccounts = 'StorageAccounts',
VirtualMachines = 'VirtualMachines',
```

And the rest are just ones we have icons for, but are hidden by default.